Makefile

```diff
@@ -37,23 +37,65 @@ else
 GOBIN=$(shell go env GOBIN)
 endif
 
-# Build actions
-all: manager injector handler
+# we define target specific variables values https://www.gnu.org/software/make/manual/html_node/Target_002dspecific.html
+injector handler: BINARY_PATH=./cli/$(BINARY_NAME)
+manager: BINARY_PATH=main.go
+
+docker-build-injector: IMAGE_TAG=$(INJECTOR_IMAGE)
+docker-build-handler: IMAGE_TAG=$(HANDLER_IMAGE)
+docker-build-manager: IMAGE_TAG=$(MANAGER_IMAGE)
+
+docker-build-ebpf:
+	docker build --platform linux/$(OS_ARCH) --build-arg ARCH=$(OS_ARCH) -t ebpf-builder-$(OS_ARCH) -f bin/ebpf-builder/Dockerfile ./bin/ebpf-builder/
+	-rm -r bin/injector/ebpf/
+	docker run --rm -v $(shell pwd):/app ebpf-builder-$(OS_ARCH)
+
+lima-push-injector lima-push-handler lima-push-manager: FAKE_FOR=COMPLETION
+
+_injector:;
+_handler:;
+_manager: generate
+
+_docker-build-injector: docker-build-ebpf
+_docker-build-handler:;
+_docker-build-manager:;
+
+# we define the template we expect for each target
+define TARGET_template
+$(1): BINARY_NAME=$(1)
```

Comment:
`$(1)` is the target name `injector|handler|manager`

Reply:
This would be helpful as a direct comment right above here

Reply:
```suggestion
# $(1) is the target name: injector|handler|manager
$(1): BINARY_NAME=$(1)
```